### PR TITLE
#83/omit young plots from prepped data

### LINKED ISSUE #83: revise unreasonable site index/tree height/age given the other two or skip young plots in data prep
## User story

```
As a user
I want unreasonable input data (such as site index of 0) to be set to reasonable values
So that every plot gets a projection
```
## Actual solution

-  a separate schema

## Implementation

Given 1 unreasonable (0, negative, etc) values in these 3 variables:

- site index
- tree height
- age 

we can solve for the missing one. Note that this also applies to missing, not just unreasonable, data. This could be config.

## Questions:

- what % of plots does this affect
- raw data (input to prep) vs prepped data (input to simulation)
- should these checks be on raw data, at end of prep, and start of simulation?

## Things to differentiate:

- missing data, erroneous data
- data erroneous because of intermediate calculations or as raw input

--------------------------------------
Other points of discussion

This user story, can be handled with #76 

```
As a user
I want the system to record and skip plots with incomplete data (null)
So that errors aren't raised / processing continues
So that I can revise the data
```

```
As a user
I want the raw and prepped data to be QA/QC'd
So that pygypsy doesn't crash with bad data
```

### PR DESCRIPTION
**Pull request by jo-tham**
closes #83

we had issues where we expected about 1% of plots to fail because of difficulty in estimating some properties for very young plots. gypsy is not calibrated on such young plots.

this enforces the young plot filter earlier in the process (in prep, instead of simulation) and makes the logic more concise. it also makes the minimum age a param in the configuration file.

### CODE REVIEW

File: pygypsy/data_prep.py [hunk 1]
```diff
@@ -211,6 +211,13 @@ def prep_standtable(data):
         tage_pl = species_dict['Pl']['tage']
         tage_sb = species_dict['Sb']['tage']
 
+        ages = [tage_aw, tage_sw, tage_pl, tage_sb]
+        if not any([age > minimum_age for age in ages]):
```
Comment (jo-tham) [inline pygypsy/data_prep.py]: this check is actually too late

we can use the plot dominant species age instead

or we have to intervene here: https://github.com/tesera/pygypsy/blob/255a559c0975d6bc9e476d3e7feb124f0e1dedc8/pygypsy/data_prep.py#L114

#76 will avoid this becoming an error, but we could save some time by solving it beforehand

lets just go with doing #76 for now and opening an new issue to solve this more effectively

Comment (juliannosambatti) [inline pygypsy/data_prep.py]: We could put an earlier check during populate_species_dict where we can check dominant species age as well and then stop here even before estimating the other ages. If it passes this check, there is this other check to verify whether the other ages are going to be too small. The problem is that we may want to estimate the variables of other species, even when one of them is too young.

Comment (jo-tham) [inline pygypsy/data_prep.py]: OK thx @juliannosambatti  - that's what I thought too.

I think we need more data from pygypsy working in the wild on real plot data. Lets keep it the way it is currently, do the #76, and then we can run on millions of plots and get more data about which ones fail and how they fail.

**Comment by jo-tham** [comment]
checked with @juliannosambatti

**Pull request merged**